docs/testing/index.md

```diff
@@ -0,0 +1,16 @@
+# Testing Guide
```

Comment:
This seems like the new entry point for testing -- could we replace the link [here](https://github.com/Automattic/wp-calypso/blob/trunk/docs/guide/index.md?plain=1#L11) to point to this page?

Reply:
Fixed - replaced the link in the Calypso Guide to point to `testing/index.md`.